edgedb-protocol/src/codec.rs

```diff
@@ -1036,6 +1045,76 @@ impl Codec for Array {
     }
 }
 
+impl Codec for Range {
+    fn decode(&self, mut buf: &[u8]) -> Result<Value, DecodeError> {
+        ensure!(buf.remaining() >= 1, errors::Underflow);
+        let flags = buf.get_u8() as usize;
+
```

Comment:
`0x01` represents an "empty range", i.e. a range with zero points, and we need to support that also.  The easiest way to test this currently is to do a `range(0, 0)`

Reply:
How do we want to expose that in our output?

Reply:
Yes, though I'm not quite sure how.  Postgres uses `empty` (literally), but I'm not sure it's a good representation.